frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx

```diff
@@ -235,7 +235,20 @@ class EmbedModalContent extends Component {
   }
 }
 
-export default connect(mapStateToProps)(EmbedModalContent);
+function getDefaultEmbeddingParams(props) {
+  const { resource, resourceParameters } = props;
+
+  return filterValidResourceParameters(
+    resource.embedding_params || {},
+    resourceParameters,
+  );
+}
+
+function filterValidResourceParameters(embeddingParams, resourceParameters) {
+  const validParameters = resourceParameters.map(parameter => parameter.slug);
+
+  return _.pick(embeddingParams, validParameters);
```

Comment:
The problem was that we don't remove the old parameter key when we rename the parameter, so let's say we rename the parameter name from `state` to `state_new` and published the `state` parameter before.

on `master` we'll save both `state` and `state_new` to `embedding_params` that's why the embedding panel still says it requires the old parameter's value (`state`) in this case.

Reply:
@WiNloSt should we do something to actually remove old parameters after a rename?

Reply:
Do you mean should we do something to remove old params that have been saved? If this PR is merged and they revisit the published question/dashboard and try to publish the question/dashboard again then the old param should be removed.

The only thing we could do to remove old parameters without involving user interactions is to migrate data on the BE. Not sure how far we want to go that route though.

Reply:
Should we maybe remove invalid parameters also when publishing, e.g. here https://github.com/metabase/metabase/blob/9bfa9296c4e871b6ab1c57d7d3a9706f628b9730/frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx#L52?

Reply:
When publishing, `embeddingParams` should already be valid. This PR addressed the issue by ensuring the state `embeddingParams` is always valid since it's been validated when we populate the state https://github.com/metabase/metabase/blob/9bfa9296c4e871b6ab1c57d7d3a9706f628b9730/frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx#L44

After that there shouldn't be a case that `embeddingParams` could contain old parameters. That's why there's no need to specifically clean/validate `embeddingParams` anywhere else.

Hope that makes sense.

Reply:
Thanks for explaining what's going on here @WiNloSt, appreciate it! Looks good to me so far!